controllers/components/index/login-form.jsx

```diff
@@ -0,0 +1,184 @@
+import React from 'react';
+
+import regex from '../helpers/regex';
+
+export default class LoginForm extends React.Component {
+    constructor( props ) {
+        super( props );
+
+        this.state = {
+            submitButton: 'Login',
+            email: '',
+            password: '',
+            errorMessage: '',
+            formIsValide: '',
+            displayError: ''
+        };
+
+        this._checkEmptyFields = this._checkEmptyFields.bind( this );
+        this._handleChange = this._handleChange.bind( this );
+        this._loginUser = this._loginUser.bind( this );
+        this._userInputIsValid = this._userInputIsValid.bind( this );
+        this._redirectSignup = this._redirectSignup.bind( this );
+    }
+
+    _handleChange( e ) {
+        const target = e.target;
+        const value = target.value;
+        const name = target.name;
+
+        this.setState({
+            [ name ]: value

```

Comment:
How can an array key be an .. array ?!

Reply:
[ES6 Computed property names syntax] (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Computed_property_names)
also in [React docs](https://facebook.github.io/react/docs/forms.html#handling-multiple-inputs)

Reply:
Interesting.. okay